### PR TITLE
chore: update MSRV to 1.74

### PR DESCRIPTION
**Pull request by 0xdevcollins**
- Update Minimum Supported Rust Version (MSRV) to 1.74
- Modify Cargo.toml to reflect new MSRV
- Update CI configuration in .github/workflows/ci.yml
- Update CHANGELOG.md with MSRV change

### CODE REVIEW

**Comment by xStrom** [comment]
We don't need to update the MSRV in order to add lints. The older toolchains will just issue a warning, which is fine.

**Comment by PoignardAzur** [comment]
I'd recommend doing multiple split PRs as you add the lints one by one and fix the clippy warnings that come up.

File: Cargo.toml [hunk 1]
```diff
@@ -7,13 +7,38 @@ edition = "2021"
...
+todo = "warn"
+unimplemented = "warn"
+unwrap_used = "warn"
+unwrap_in_result = "warn"
+enum_glob_use = "warn"
+use_self = "warn"
+integer_arithmetic = "warn"
+match_on_vec_items = "warn"
+imprecise_flops = "warn"
+
```
Comment (DJMcNab) [inline Cargo.toml:41]: Hmm, where did this set of lints come from? This isn't as discussed in #369

For reference, the main guidance was recently published in https://linebender.org/wiki/canonical-lints/

**Comment by waywardmonkeys** [comment]
Thanks for this contribution.

Lint updates have already landed no on `main` and we aren't going to bump the MSRV at this time.

**Pull request closed without merging**